### PR TITLE
PRMT-2730

### PR DESCRIPTION
**Pull request by martin-nhs**
- Added functionality to send a delete request to the ehr repo once we receive an integration acknowledgement.
- Added unit and integration tests.

### CODE REVIEW

File: src/services/handler/acknowledgement-handler.js [hunk 1]
```diff
@@ -4,21 +4,50 @@ import { setCurrentSpanAttributes } from "../../config/tracing";
...
+const handlePositiveIntegrationAcknowledgement = async (nhsNumber, conversationId) => {
+  const usefulDetails = `for Conversation ID ${conversationId}, and NHS number ${nhsNumber}.`;
+  logInfo(`Positive integration acknowledgement received ${usefulDetails}`);
+  logInfo(`Sending delete request to ehr out repository ${usefulDetails}`);
+
+  await sendDeleteRequestToEhrRepo(nhsNumber, conversationId);
+};
+
+const handleNegativeIntegrationAcknowledgement = () => {
+  logInfo(`Negative integration acknowledgement received.`);
```
Comment (AndyFlintNHS) [inline src/services/handler/acknowledgement-handler.js:44]: James Insley has confirmed that we want negative acknowledgements to also trigger the deletion of the record. We will need to establish at a later date what happens in the event that no integration is ever received.

**Review by AndyFlintNHS — CHANGES_REQUESTED** [review]

File: src/services/database/__tests__/registration-request-repository.integration.test.js [hunk 2]
```diff
@@ -3,34 +3,34 @@ import ModelFactory from '../../../models';
...
+  });
+
+  afterAll(async () => {
     await ModelFactory.sequelize.close();
   });
 
   it('should retrieve the registration request by conversation id', async () => {
     // given
     const conversationId = '22a748b2-fef6-412d-b93a-4f6c68f0f8dd';
+    const messageId = uuid();
```
[the remark this answers is not in the document]
Comment (martin-nhs) [reply src/services/database/__tests__/registration-request-repository.integration.test.js]: Comment addressed!

File: src/services/database/__tests__/registration-request-repository.integration.test.js [hunk 3]
```diff
@@ -108,4 +98,36 @@ describe('Registration request repository', () => {
...
+      const nhsNumber = '1247415214'
+      const status = Status.REGISTRATION_REQUEST_RECEIVED
+
+      // when
+      await createRegistrationRequest(conversationId, nhsNumber, messageId, odsCode);
+
+      const foundRecord = await registrationRequestExistsWithMessageId(messageId);
+
+      // then
+      expect(foundRecord).not.toBeNull();
```
[the remark this answers is not in the document]
Comment (martin-nhs) [reply src/services/database/__tests__/registration-request-repository.integration.test.js]: Very good spot! Comment addressed.

File: src/services/handler/acknowledgement-handler.js (hunk 1 above)
Comment (martin-nhs) [reply src/services/handler/acknowledgement-handler.js]: Comment addressed!

**Review by AndyFlintNHS — APPROVED** [review]

**Pull request merged**